src/components/GoogleCalendarButton.js

```diff
@@ -17,50 +17,97 @@ const useStyles = makeStyles({
+
+  const { enqueueSnackbar } = useSnackbar();
+  const message = 'Classes successfully exported to Google Calendar';
+  const exportSuccess = () => {
+    enqueueSnackbar(message, {
+      variant: 'success',
+    });
+  };
+  const events = sections.map((section) => {
+    const dow = section.event.dow;
+    const date = dateOfTheFirstClass[dow];
+    const startTime = `${date + section.event.start.hour}:${section.event.start.minute}:00`;
+    const endTime = `${date + section.event.end.hour}:${section.event.end.minute}:00`;
+    const event = {
+      summary: `${section.subjectId} ${section.courseId}`,
+      description: section.name,
+      location: section.event.location,
+      start: {
+        dateTime: startTime,
+        timeZone: 'America/Chicago',
       },
-    );
-    return events;
+      end: {
+        dateTime: endTime,
+        timeZone: 'America/Chicago',
+      },
+      recurrence: [
+        'RRULE:FREQ=WEEKLY;COUNT=10',
+      ],
+    };
+    return event;
+  });
+
+  const initAuth = () => {
+    window.gapi.load('auth2', () => {
+      window.gapi.auth2.init({
+        client_id: clientID,
+      }).then(() => {
+        window.gapi.load('client', () => {
+          window.gapi.client.setApiKey(APIKey);
+          window.gapi.client.load('https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest')
+            .then(() => { console.log('GAPI client loaded for API'); },
+              (err) => { console.error('Error loading GAPI client for API', err); });
+        });
+        window.gapi.auth2.getAuthInstance()
+          .signIn({ scope: 'https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/calendar.events'})
+          .then(() => {
+            events.forEach(
+              (event) => {
+                const newEvent = {
+                  calendarId: 'primary',
+                  resource: event,
+                };
+                window.gapi.client.calendar.events.insert(newEvent);
+              },
+            );
+          },
+          (err) => { console.error('Error signing in', err); });
+      }).then(() => {
+        exportSuccess();
+      });
+    });
+  };
+
+  const loadGapiAndAfterwardsInitAuth = () => {
+    const script = document.createElement('script');
+    script.src = 'https://apis.google.com/js/platform.js';
+    script.async = true;
+    script.defer = true;
+    script.onload = initAuth;
+    const meta = document.createElement("meta");
+    meta.name = 'google-signin-client_id';
+    meta.content = '%REACT_APP_GOOGLE_ID_OF_WEB_CLIENT%';
+    document.head.appendChild(meta);
+    document.head.appendChild(script);
   };
 
   return (
     <div>
       <Tooltip title="Export to Google Calendar">
         <IconButton aria-label="GoogleCalendar">
-          <EventIcon onClick={handleClick} className={classes.icon} />
+          <EventIcon onClick={loadGapiAndAfterwardsInitAuth} className={classes.icon} />
```

Comment:
The `onClick` should actually be on the `IconButton` – there's a bunch of warning statements in console otherwise, because apparently it breaks in Firefox!